### DISCUSSION TITLE
Privacy in Fabric?

### DISCUSSION THREAD

**Discussion by geologic9** [danielmiessler/Fabric#1009] [General]
Hey all, sorry if this is a very silly question.

I installed GO and saw it pulled a bunch of telemetry related links from google et al. I'm mostly wanting to to use Fabric to reduce dependency on centralized AI like ChatGPT, so privacy is first and foremost. I've done some reading and it looks like GO has some telemetry built in, is this an issue or can it be disabled? (Im a bit new to this, so even a point in the right direction would be appreciated).

Thanks everyone.

**Comment by yousecjoe** [comment]
> Hey all, sorry if this is a very silly question.
> 
> I installed GO and saw it pulled a bunch of telemetry related links from google et al. I'm mostly wanting to to use Fabric to reduce dependency on centralized AI like ChatGPT, so privacy is first and foremost. I've done some reading and it looks like GO has some telemetry built in, is this an issue or can it be disabled? (Im a bit new to this, so even a point in the right direction would be appreciated).
> 
> Thanks everyone.

This applies to everything you install. 

It also applies to security and privacy in general. 

First, identify the asset.
Second, monitor the traffic flow. 
Third, block what you do not want to communicate.

It really is that simple. 

This is part of the Zero Trust strategy and maturity model. 

It's been possible forever, though. Nothing prevents you from blocking whatever source and destinations you like. 

You do not need an internet connection to run a Go app. Go is capable of running air-gapped. So there is no issue here. It's entirely up to you to manage your network security and data privacy. 

For example, I run local genAI solutions and restrict them so they can not communicate outside a micro segmentation, nano segmentation, protect surface, etc. You can call it whatever you want. The name doesn't matter. The process is what matters most.

**Reply by HippyOnTheMoon999** [reply]
True

**Comment by digitalextremist** [comment]
Telemetry is not a default for everything you install.

A language with telemetry baked into it by default is a wrong philosophy. Working to reinforce wise behavior does not involve fighting telemetry first, but fighting the design philosophy, then its implementation.

Going one further:

`go` does not have `go uninstall` and [there is no uninstall procedure](https://github.com/danielmiessler/fabric/issues/1061) given for `fabric` alongside installation.

One might say "rtfm and notice that it means `rm -rf <something>`" ... but why? Software that is a one-way install is presumptuous and bad form, by a lot. Then one will also notice there is a ton of cruft also installed and the best solution just becomes `rm -rf ~/go` ... Because again, bad philosophy. How many of those 😬 moments need to happen before we throw out a tool on principle?

There is no deeply positive feeling that comes across from `fabric`; only the mantras and brand that _ought_ to give that sense: of privacy, performance, quality, and sovereignty... but the code smell is saying "watch your back" more than unleashing freedom.

Why let something into the very fabric of my workflow when its default philosophies are cringe-worthy at best?
